lib/runners/processor/metrics_fileinfo.rb

```diff
@@ -20,6 +20,11 @@ def use_git_metadata_dir?
     end
 
     def analyze(changes)
+      # Generate pre-computed cache for Git metadata access (https://git-scm.com/docs/git-commit-graph)
+      # This improves the performance of access to Git metadata for a large repository.
+      # You can see the efficacy here: https://github.com/sider/runners/issues/2028#issuecomment-776534408
+      capture3!("git", "commit-graph", "write", "--reachable", "--changed-paths")
```

Comment:
[question] Does it make sense to add `--no-progress` to reduce trace size?

```suggestion
      capture3!("git", "commit-graph", "write", "--reachable", "--changed-paths", "--no-progress")
```

Reply:
I thought as you said, but I found this description, so it might be unnecessary.

https://git-scm.com/docs/git-commit-graph
>  If neither is specified, progress is shown if standard error is connected to a terminal.

Reply:
@ybiquitous 
Actually, output (stderr) is only a line. But it may be changed in the future. Thank you for your suggestion.
```
$ ../git/git-commit-graph write --changed-paths --reachable
Computing commit changed paths Bloom filters: 100% (74050/74050), done.
$ 
```

@yykamei 
Runner collects `stderr` alike `stdout`, so we can reduce trace log size with `--no-progress` option. Is it right?
(Perhaps, I maybe misunderstand your intension...)

Reply:
No, Git internally seems to use [tcgetpgrp(3)](https://man7.org/linux/man-pages/man3/tcgetpgrp.3.html), so it is a matter of the existence of a terminal. I tried with `--no-progress` and without `--no-progress`, and got the same results.

```console
mkdir without-no-progress/
cd without-no-progress/
git init
git remote add origin git@github.com:DefinitelyTyped/DefinitelyTyped.git
git fetch origin
irb
require 'open3'
Open3.capture3('git', 'commit-graph', 'write', '--reachable', '--changed-paths')
=> ["", "", #<Process::Status: pid 28239 exit 0>]

mkdir with-no-progress/
cd with-no-progress/
git init
git remote add origin git@github.com:DefinitelyTyped/DefinitelyTyped.git
git fetch origin
irb
require 'open3'
Open3.capture3('git', 'commit-graph', 'write', '--reachable', '--changed-paths', '--no-progress')
=> ["", "", #<Process::Status: pid 28239 exit 0>]
```

Reply:
@yykamei

I got it. Runner trace the output of a target process but don't create any psuedo terminal for the process. So, `git commit-graph` doesn't output progress even if with `--progress` option. Thanks for your suggestion and experiments 🙏  I'll remove the option in the next refactoring opportunity.